### PR TITLE
first stab at fixing recs size errors

### PR DESCRIPTION
**Pull request by keener101**
I expect to get some substantial 'fix this' feedback on this push.

The fix itself seems simple enough - just expand the checks to turn off recommendations to also include `0` and `False` (in addition to the previously checked `None`), and then after that, set `recommend=None` if `True` is passed in. 

I feel a bit out of my depth currently with the test classes when they get this complex. I copied the pattern from other tests that utilized `MultiEval`. 

`test_sweep_norecs` creates a `MultiEval` with `recommend = None` and makes sure no `recommendations.parquet` file is created. It then *repeats* the process with `recommend = False`. I wasn't sure if you'd want this or if it's overkill (especially given how slow these tests are). I could have also added one for `recommend = 0`, but didn't.

There's a bunch of additional `assert` lines that aren't really needed to test this - but I figured I'd leave them in during the first submission in case you'd prefer to check them regardless. 

`test_sweep_allrecs` simply makes sure a `recommendations.parquet` file is created when `recommend=True` is passed. **However it does not check if full recommendations are generated**. I wasn't sure how I would check this - I imagine it would work with an ``assert len(recs) == X`` check, but I'm not sure exactly what that size would be. (I have some confusion about ``ml_pandas.renamed.ratings`` and where exactly it is coming from).

### CODE REVIEW

File: lenskit/batch/_multi.py [hunk 1]
```diff
@@ -291,9 +291,12 @@ def _predict(self, rid, algo, test):
         return preds, watch.elapsed()
 
     def _recommend(self, rid, algo, test, candidates):
-        if self.recommend is None:
+        if self.recommend is None or self.recommend is False or self.recommend is 0:
```
Comment (mdekstrand) [inline lenskit/batch/_multi.py]: why not just `if not self.recommend`? That covers all the falsey values.

File: lenskit/batch/_multi.py [hunk 2]
```diff
@@ -291,9 +291,12 @@ def _predict(self, rid, algo, test):
...
             return None, None
 
+        if self.recommend is True:
+            self.recommend = None
```
Comment (mdekstrand) [inline lenskit/batch/_multi.py]: This is going to change the recommend list size for the next run-through, not what we want. We can instead do a local variable:

```python
if not self.recommend:
    return stuff
elif self.recommend is True:  # special value True means unlimited
    nrecs = None
else:  # recommend has rec size
    nrecs = self.recommend
```

File: lenskit/batch/_multi.py (hunk 1 above)
Comment (keener101) [reply lenskit/batch/_multi.py]: I... did not realize that. How elegant.

**Comment by mdekstrand** [comment]
To link things: this addresses #83.

**Pull request merged**